src/litdata/streaming/item_loader.py

```diff
@@ -360,6 +360,13 @@ def delete(self, chunk_index: int, chunk_filepath: str) -> None:
                 del self._mmaps[chunk_index]
             os.remove(chunk_filepath)
 
+    def close(self, chunk_index: int) -> None:
+        """Release the memory-mapped file for a specific chunk index."""
+        if chunk_index in self._mmaps:
```

Comment:
We need to close the self._mmaps before deleting it. However, this used to cause some failures.

Reply:
It seems `np.memmap` does not provide a direct `close` method.

> [Currently, there is no API to close the underlying mmap. It is tricky to ensure the resource is actually closed, since it may be shared between different memmap instances.](https://numpy.org/doc/stable/reference/generated/numpy.memmap.html#numpy.memmap)

However, in this issue (https://github.com/numpy/numpy/issues/13510), I found a way to close it with the following:

```python
self._mmaps[chunk_index]._mmap.close()
```

I'll try to implement this and check for any potential errors.